executor/executor.go

```diff
@@ -407,59 +411,87 @@ func (e *ShowDDLJobsExec) Open(ctx context.Context) error {
 	if e.jobNumber == 0 {
 		e.jobNumber = admin.DefNumHistoryJobs
 	}
-	historyJobs, err := admin.GetHistoryDDLJobs(txn, int(e.jobNumber))
+
+	m := meta.NewMeta(txn)
+	historyJobIter, err := m.GetLastHistoryDDLJobsIterator()
+
+	//historyJobs, err := admin.GetHistoryDDLJobs(txn, int(e.jobNumber))
```

Comment:
one question: I find the jobs returned is still the same, same slice size...
So the key to avoid oom are: 1.avoid sort & 2. kv iterator eliminate memory allocation?

Reply:
Yes, it mostly is eliminate the large memory allocation.

For example, before this PR, If we need to get 1 million history DDL jobs, we need to allocate a slice with 1 million size.

Reply:
got